src/compiler/factory.ts

```diff
@@ -667,17 +667,18 @@ namespace ts {
         return <KeywordTypeNode>createSynthesizedNode(kind);
     }
 
-    export function createTypePredicateNode(parameterName: Identifier | ThisTypeNode | string, type: TypeNode) {
+    export function createTypePredicateNode(assertsModifier: AssertsToken | undefined, parameterName: Identifier | ThisTypeNode | string, type: TypeNode | undefined) {

```

Comment:
this is a breaking API change.
typically there is a new overload added to maintain backwards compatibility. the old signature can be marked as deprecated right away and could be removed later.